src/reducers/ui.js

```diff
@@ -153,10 +159,30 @@ export function getFrameworkGroupingState(state: OuterState): boolean {
   return state.ui.get("frameworkGroupingOn");
 }
 
-export function getShownSource(state: OuterState): boolean {
+export function getShownSource(state: OuterState): Source {
   return state.ui.get("shownSource");
 }
 
+export function getSourceFromPrettySource(
+  state: OuterState,
+  baseSource: Source
+): Source | null {
+  if (!baseSource) {
+    return null;
+  }
+  if (!baseSource.isPrettyPrinted) {
+    return baseSource;
+  }
+
+  const sources = getRelativeSources(state);
+  const id = Object.keys(sources).find(sourceId => {
+    return (
+      getRawSourceURL(sources[sourceId].url) === getRawSourceURL(baseSource.url)
+    );
+  });
```

Comment:
can we use a selector from `reducers/sources`? If you need to move this selector into the selectors directory that's cool too